base/expr.jl

```diff
@@ -2,12 +2,12 @@
 
 ## symbols ##
 
-symbol(s::Symbol) = s
-symbol(s::ASCIIString) = symbol(s.data)
-symbol(s::UTF8String) = symbol(s.data)
-symbol(a::Array{UInt8,1}) =
+Symbol(s::Symbol) = s
+Symbol(s::ASCIIString) = Symbol(s.data)
+Symbol(s::UTF8String) = Symbol(s.data)
+Symbol(a::Array{UInt8,1}) =
     ccall(:jl_symbol_n, Ref{Symbol}, (Ptr{UInt8}, Int32), a, length(a))
-symbol(x...) = symbol(string(x...))
+Symbol(x...) = Symbol(string(x...))
```

Comment:
@vtjnash can confirm whether that's what he meant, but at the minimum you should change the test I referred to. For example, replace `Symbol` with `sum()` (be sure to change the error message check too).

Then, please squash the changes into 2 commits. You can do that using `git rebase -i master`, reordering the commits as needed, and then writing `squash` for those you want to merge with the preceding commit. Then, do `git push jeffreysarnoff-dev +jas/deprecate_symbol` to update the PR.